### PR TITLE
Proxy e2e test

### PR DESCRIPTION
**Pull request by jhadvig**
Test scenario: 
- e2e test for setting `NoProxy` & `HTTPProxy` & `HTTPSProxy` to the Proxy CR and checking the new rolledout console deployment pod for corresponding environment variables (`NO_PROXY` & `HTTP_PROXY` & `HTTPS_PROXY`) 

/assign @benjaminapetersen

### CODE REVIEW

File: test/e2e/proxy_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		},
+		"clusterVars": {
+			{Name: "NO_PROXY", Value: clusterProxyConfig.NoProxy, ValueFrom: nil},
+			{Name: "HTTP_PROXY", Value: clusterProxyConfig.HTTPProxy, ValueFrom: nil},
+			{Name: "HTTPS_PROXY", Value: clusterProxyConfig.HTTPSProxy, ValueFrom: nil},
+		},
+	}
+
+	consoleDeployment, err := framework.GetConsoleDeployment(client)
+	if err != nil {
```
Comment (benjaminapetersen) [inline test/e2e/proxy_test.go]: So can this run without the Proxy CR?  This is a sufficient test?

Comment (benjaminapetersen) [reply test/e2e/proxy_test.go]: I could see on our end that passing the envs to the deployment is sufficient.

But do we then need a test in `console` validating that it responds through the proxy?

Comment (jhadvig) [reply test/e2e/proxy_test.go]: We probably should. Lets talk about that :)

**Comment by jhadvig** [comment]
@benjaminapetersen updated the PR, cause I noticed it was flaking. Should be fixed now. PTAL

**Comment by jhadvig** [comment]
@benjaminapetersen thats the flake I've seen locally every N-th run of the test.. but with changes now I was not able to reproduce...investigating

**Comment by benjaminapetersen** [comment]
/lgtm

I think this does a good job proving that we do what we are supposed to do when given a proxy config.  

Now, I'm still curious about a proxy test that validates that the cluster works and the console/components is/are responsive through the proxy.  Was that still something being looked into?

**Pull request merged**